include/dsn/c/api_layer1.h

```diff
@@ -541,26 +542,25 @@ typedef enum dsn_msg_parameter_type_t
     MSG_PARAM_NONE = 0,           ///< nothing  
 } dsn_msg_parameter_type_t;
 
-enum dsn_msg_serialize_format
+typedef enum dsn_msg_serialize_format
 {
-    DSF_THRIFT_BINARY,
-    DSF_THRIFT_COMPACT,
-    DSF_THRIFT_JSON,
-    DSF_PROTOC_BINARY,
-    DSF_PROTOC_JSON,
-
-    DSF_COUNT,
```

Comment:
I think count is useful (e.g., callbacks[dsf-count], and i'm not sure the reason you did the change.